src/vm/eetoprofinterfaceimpl.cpp

```diff
@@ -2056,6 +2057,16 @@ HRESULT EEToProfInterfaceImpl::EnsureProfilerDetachable()
         return CORPROF_E_IRREVERSIBLE_INSTRUMENTATION_PRESENT;
     }
 
+    if (m_fModifiedRejitState)
```

Comment:
The way I have it written, I don't believe it is identical. In SetEventMask, I am not currently preventing profilers from disabling` COR_PRF_ENABLE_REJIT`, so it would be possible for a profiler to detach after setting the event mask to not include `COR_PRF_ENABLE_REJIT` if there was only a check for `CORProfilerEnableRejit()`.

So, either we need the check that I added to see if it's ever been enabled, or prevent SetEventMask from turning off `COR_PRF_ENABLE_REJIT` once it's been enabled. I'm not particularly partial to either one, this is just the one I chose.

Reply:
You are absolutely right, lets leave like this.